apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -129,16 +129,23 @@ handle_request('GetAccountBalance', Req, _Context) ->
               {ok, PK} = aec_keys:pubkey(),
               PK;
           PK ->
-              base64:decode(PK)
+              try base64:decode(PK)
+              catch error:{badarg, _} -> not_base64_encoded
+              end
       end,
-    {ok, LastBlock} = aec_chain:top(),
-    Trees = aec_blocks:trees(LastBlock),
-    AccountsTree = aec_trees:accounts(Trees),
-    case aec_accounts:get(Pubkey, AccountsTree) of
-        {ok, #account{balance = B}} ->
-            {200, [], #{balance => B}};
+    case Pubkey of
+        not_base64_encoded ->
+            {400, [], #{reason => <<"Invalid address">>}};
         _ ->
```

Comment:
This open case is fragile as the atom of the clause above may be mistyped by mistake. I suggest adding guard (e.g. `when is_binary(...)`).

Reply:
One can never be too assertive, I've pushed a change for this